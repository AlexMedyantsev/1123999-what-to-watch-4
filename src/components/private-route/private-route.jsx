import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {SERVER_ROUTE} from "../../utils/consts.js";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {getAuthorizationStatus} from "../../reducer/user/selectors.js";

const PrivateRoute = (props) => {
  return (
    props.authorizationStatus === AuthorizationStatus.AUTH ? <Route {...props}/> : <Redirect to={SERVER_ROUTE.LOGIN}/>
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => {
  return {
    authorizationStatus: getAuthorizationStatus(state),
  };
};

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
