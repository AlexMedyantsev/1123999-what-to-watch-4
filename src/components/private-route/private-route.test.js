import React from "react";
import rerender from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {MemoryRouter} from "react-router";
import NameSpace from "../../reducer/name-space.js";

import {PrivateRoute} from "./private-route.jsx";

const mockStore = configureStore([]);

describe(`MoviePage test`, () => {
  it(`Should MoviePage component render correctly`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        movies: [{
          image: `a`,
          posterSrc: `a`,
          bgSrc: `a`,
          title: `Name`,
          genre: `Genre`,
          year: 2014,
          score: 8.9,
          level: `level`,
          movieLink: `link`,
          previewLink: `preview-link`,
          scoresCount: 220,
          description: `description`,
          director: `director`,
          starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
          runTime: `01h30m`,
          link: `movie-page.html`,
          id: 1,
          backgroundColor: `background_color`,
          isFavorite: true,
        },
        {
          image: `a`,
          posterSrc: `a`,
          bgSrc: `a`,
          title: `Name`,
          genre: `Genre`,
          year: 2014,
          score: 8.9,
          level: `level`,
          movieLink: `link`,
          previewLink: `preview-link`,
          scoresCount: 220,
          description: `description`,
          director: `director`,
          starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
          runTime: `01h30m`,
          link: `movie-page.html`,
          id: 2,
          backgroundColor: `background_color`,
          isFavorite: true,
        }],
        promoMovie: {}
      },
      [NameSpace.CONDITION]: {
        currentGenre: ``,
        countMoviesShow: ``,
      },
      [NameSpace.USER]: {
        authorizationStatus: `AUTH`,
      }
    });
    const tree = rerender
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <PrivateRoute
                authorizationStatus={`AUTH`}
              />
            </MemoryRouter>
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          }
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
