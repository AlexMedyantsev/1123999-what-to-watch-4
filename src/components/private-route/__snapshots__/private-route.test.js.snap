// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage test Should MoviePage component render correctly 1`] = `null`;
